app/pages/Dashboard/Households/HouseholdList/HouseholdList.tsx

```diff
@@ -6,20 +6,87 @@ import {
   showNotification,
   Table,
 } from '@camp/design';
+import { householdColumnHelper } from '@camp/domain';
 import { errorMessages, messages } from '@camp/messages';
+import { AppRoute } from '@camp/router';
 import { createTestAttr } from '@camp/test';
 import { isEmpty, isNull } from '@fullstacksjs/toolbox';
+import { Group } from '@mantine/core';
+import type { SortingState } from '@tanstack/react-table';
+import {
+  getCoreRowModel,
+  getSortedRowModel,
+  useReactTable,
+} from '@tanstack/react-table';
+import { useState } from 'react';
 
+import { InformationBadge, SeverityBadge } from '../../../../components';
 import { CreateHouseholdButton } from '../CreateHousehold';
+import {
+  HouseholdActionButton,
+  householdActionIds as actionIds,
+} from '../HouseholdActionButton';
 import { HouseholdEmptyState } from '../HouseholdEmptyState';
 import * as ids from './HouseholdList.ids';
+import { HouseholdTableColumn } from './HouseholdTableColumn';
 import { HouseholdTableRow } from './HouseholdTableRow';
 
+const t = messages.households.list;
+
 export const HouseholdList = () => {
-  const t = messages.households.list;
-  const { data, loading, error } = useHouseholdListQuery();
+  const [sorting, setSorting] = useState<SortingState>([]);
+
+  const { data, loading, error } = useHouseholdListQuery({
+    variables: { orderBy: sorting },
+  });
   const households = data?.household;
 
+  const columns = [
+    householdColumnHelper.display({
+      header: t.table.columns.order,
+      cell: order => order.row.index + 1,
+    }),
+    householdColumnHelper.accessor('name', {
+      header: t.table.columns.name,
+      cell: name => name.getValue(),
+    }),
+    householdColumnHelper.accessor('isCompleted', {
+      id: 'status',
+      header: t.table.columns.status,
+      cell: status => (
+        <InformationBadge status={status.getValue() ? 'completed' : 'draft'} />
+      ),
+    }),
+    householdColumnHelper.accessor('severity', {
+      header: t.table.columns.severity,
+      enableSorting: true,
+      cell: props => (
+        <Group position="apart">
+          <SeverityBadge severity={props.getValue()} />
+          <HouseholdActionButton
+            to={AppRoute.householdDetail}
+            params={{ id: props.row.original.id }}
+            household={props.row.original}
+            menuButtonId={actionIds.actionButton}
+            menuId={actionIds.actionMenu}
+          />
+        </Group>
+      ),
+    }),
+  ];
```

Comment:
Let's move it out of the component to not create a new instance on every render.

Reply:
like where?

Reply:
@erfan-goodarzi just outside the component, before `export const HouseholdList = () => {`